src/expr/src/relation/func.rs

```diff
@@ -787,6 +853,7 @@ impl AggregateFunc {
             AggregateFunc::ListConcat { .. } => Datum::empty_list(),
             AggregateFunc::RowNumber { .. } => Datum::empty_list(),
             AggregateFunc::DenseRank { .. } => Datum::empty_list(),
+            AggregateFunc::Lag { .. } => Datum::empty_list(),
```

Comment:
I might look into this function with more attention in the future, as I don't think our window functions actually have such identity, but since the only place where this function is used is to check whether the `identity_datum` is null, it's not really pressing.

Reply:
If including an empty list changes the output of the result, I believe it's not really a...monoid (or some such). Seems troubling!

Reply:
It's not! None of the scalar/ranking/value window functions are monoids (they are all 1:1, they just happen to depend on data of the entire window frame), and they aren't truly reductions either. In HIR they are not `AggregateFunc`s, so this is not an issue there (where this is really important, since the HIR version of `identity_datum` is used to implement `FILTER`), but in MIR they are `AggregateFunc`s, so we get this code block.

E.g. take `row_number()`: if you add another row, you get another row number (and another output row, for that matter), no matter what the contents of the new row are. There is simply no way to change that property.

Also, `FILTER` is not supported for non-aggregate window functions in Postgres, so not having `identity_datum` for them doesn't change anything.